lib/src/rules/no_leading_underscores_for_local_identifiers.dart

```diff
@@ -113,6 +114,13 @@ class _Visitor extends SimpleAstVisitor<void> {
     }
   }
 
+  @override
+  void visitFunctionDeclaration(FunctionDeclaration node) {
+    if (node.parent is FunctionDeclarationStatement) {
```

Comment:
Consider visiting `FunctionDeclarationStatement` and unconditionally invoking `checkIdentifier(node.functionDeclaration.name)`. It should be more performant because it will avoid a type check.

Reply:
Excellent.  Thanks!